src/ai/backend/agent/docker/utils.py

```diff
@@ -104,10 +107,17 @@ async def install_latest(self) -> None:
         with gzip.open(self.img_path, "rb") as reader:
             proc = await asyncio.create_subprocess_exec(
                 *["docker", "load"],
-                stdin=cast(BinaryIO, reader),
+                stdin=subprocess.PIPE,
                 stdout=subprocess.DEVNULL,
                 stderr=subprocess.PIPE,
             )
+            assert proc.stdin
+            while True:
+                chunk = reader.read(IMAGE_CHUNK_SIZE)
+                if not chunk:
+                    break
+                proc.stdin.write(chunk)
+            await proc.stdin.drain()
```

Comment:
I think it will be better to drain at each chunk, and increase the chunk size a little bit (at least megabytes).